lib/task_helpers.rb

```diff
@@ -1,213 +1,106 @@
 require 'elasticsearch'
+require_relative '../models/comment'
+require_relative '../models/comment_thread'
 
 module TaskHelpers
   module ElasticsearchHelper
     LOG = Logger.new(STDERR)
+    INDEX_MODELS = [Comment, CommentThread].freeze
+    INDEX_NAMES = [Comment.index_name, CommentThread.index_name].freeze
 
-    # Creates a new index and loads data from the database.  If an alias name
-    # is supplied, it will be pointed to the new index and catch up will be
-    # called both before and after the alias switch..
-    #
-    # Returns the name of the newly created index.
+    # Creates new indices and loads data from the database.
     #
     # Params:
-    # +alias_name+:: (optional) The alias to point to the new index.
     # +batch_size+:: (optional) The number of elements to index at a time. Defaults to 500.
-    # +sleep_time+:: (optional) The number of seconds to sleep between batches. Defaults to 0.
     # +extra_catchup_minutes+:: (optional) The number of extra minutes to catchup. Defaults to 5.
-    def self.rebuild_index(alias_name=nil, batch_size=500, sleep_time=0, extra_catchup_minutes=5)
-      initial_start_time = Time.now
-      index_name = create_index()
-
-      [Comment, CommentThread].each do |model|
+    def self.rebuild_indices(batch_size=500, extra_catchup_minutes=5)
+      create_indices
+      INDEX_MODELS.each do |model|
         current_batch = 1
-        model.import(index: index_name, batch_size: batch_size) do |response|
-            batch_import_post_process(response, current_batch, sleep_time)
+        model.import(index: model.index_name, batch_size: batch_size) do |response|
+            batch_import_post_process(response, current_batch)
             current_batch += 1
         end
       end
 
-      if alias_name
-        # Just in case initial rebuild took days and first catch up takes hours,
-        # we catch up once before the alias move and once afterwards.
-        first_catchup_start_time = Time.now
-        adjusted_start_time = initial_start_time - (extra_catchup_minutes * 60)
-        catchup_index(adjusted_start_time, index_name, batch_size, sleep_time)
+      first_catchup_start_time = Time.now
+      adjusted_start_time = first_catchup_start_time - extra_catchup_minutes * 60
+      catchup_indices(adjusted_start_time, batch_size)
 
-        move_alias(alias_name, index_name, force_delete: true)
```

Comment:
Do we understand what problem the alias was solving?  Why is it we no longer need it?

Reply:
As I see from doc by default alias was added to the index with a timestamp. This will enable to swap out indices (e.g. rebuild_index) without having to take downtime or modify code with a new index name, so we will produce a lot of indices. Current app is working with actual index, but I think that it's overhead of memory. I think that it is much easier to rebuild indices with actual data than produce new indices.
if i misunderstand issue which aliases resolve let me know, please.

Reply:
Doesn't this mean that we can no longer rebuild the indices without downtime, or at least, without the search returning no results/incomplete results for a while during the rebuild?  I'd want to run this by some folks here who may be more familiar with our edx.org forums and how long they take to rebuild - from some of the comments I see in the code, it sounds like it may take quite some time, during which search would be broken for users.

Reply:
Yes, you are right. Does it mean that alias functional should be returned?

Reply:
I think so, yes.  From what we heard in our Slack thread earlier today, it seems like using an alias for reindexing is considered a best practice, and from a practical point of view, it sounds like we need to ensure that there's no down time while re-indexing occurs, which the prior code seems to have accomplished.  The edX devops team indicated that reindexing for edx.org can take _hours_ to finish, which is definitely an unacceptably long time to have an incomplete/non-existent index.